RecoLocalTracker/SiPixelRecHits/src/PixelCPEGeneric.cc

```diff
@@ -518,46 +570,147 @@ float PixelCPEGeneric::generic_position_formula(int size,                    //!
 //!  and the inner cluster charge, projected in x and y.
 //-----------------------------------------------------------------------------
 void PixelCPEGeneric::collect_edge_charges(ClusterParam& theClusterParamBase,  //!< input, the cluster
-                                           int& Q_f_X,                         //!< output, Q first  in X
-                                           int& Q_l_X,                         //!< output, Q last   in X
-                                           int& Q_f_Y,                         //!< output, Q first  in Y
-                                           int& Q_l_Y                          //!< output, Q last   in Y
-                                           ) const {
+                                           bool& del_Xmin,
+                                           bool& del_Xmax,
+                                           bool& del_Ymin,
+                                           bool& del_Ymax,
+                                           int& Q_f_X,  //!< output, Q first  in X
+                                           int& Q_l_X,  //!< output, Q last   in X
+                                           int& Q_f_Y,  //!< output, Q first  in Y
+                                           int& Q_l_Y,  //!< output, Q last   in Y
+                                           double xtalk[4],
+                                           int induced_thre,
+                                           bool correct_xtalk) const {
   ClusterParamGeneric& theClusterParam = static_cast<ClusterParamGeneric&>(theClusterParamBase);
 
   // Initialize return variables.
-  Q_f_X = Q_l_X = 0.0;
-  Q_f_Y = Q_l_Y = 0.0;
+  del_Xmin = del_Xmax = false;
+  del_Ymin = del_Ymax = false;
+  Q_f_X = Q_l_X = 0;
+  Q_f_Y = Q_l_Y = 0;
+
+  // xtalk matrix is defined as a symmetric 2x2 matrix [ [1-xtalk,xtalk] , [xtalk,1-xtalk] ] which
+  // has been applied to pair pixels and is to be corrected with its inverse
+  // the four values in xtalk[4] correspond to the coupling between a pixel in an odd row and column to its next column, next row, previous column, previous row
+  float xtalk_matrix_determinant[4];
```

Comment:
4 should be a named constant to ensure consistency among all the arrays/operations/loops here